tests/Crypto/tst_cryptorequests/tst_cryptorequests.cpp

```diff
@@ -392,29 +396,55 @@ void tst_cryptorequests::validateCertificateChain()
     QSKIP("TODO - certificate validation not yet implemented!");
 }
 
+void tst_cryptorequests::signVerify_data()
+{
+    QTest::addColumn<CryptoManager::Algorithm>("algorithm");
+    QTest::addColumn<CryptoManager::DigestFunction>("digestFunction");
+
+    QTest::newRow("RSA + SHA256") << CryptoManager::AlgorithmRsa << CryptoManager::DigestSha256;
+    QTest::newRow("RSA + SHA512") << CryptoManager::AlgorithmRsa << CryptoManager::DigestSha512;
+    QTest::newRow("RSA + MD5") << CryptoManager::AlgorithmRsa << CryptoManager::DigestMd5;
+    QTest::newRow("EC + SHA256") << CryptoManager::AlgorithmEc << CryptoManager::DigestSha256;
+    QTest::newRow("EC + SHA512") << CryptoManager::AlgorithmEc << CryptoManager::DigestSha512;
+}
+
+static inline KeyPairGenerationParameters *getKeyPairGenerationParameters(CryptoManager::Algorithm algorithm)
+{
+    switch (algorithm)
+    {
+    case CryptoManager::AlgorithmRsa:
+        return new RsaKeyPairGenerationParameters();
+    case CryptoManager::AlgorithmEc:
+        return new EcKeyPairGenerationParameters();
+    default:
+        throw std::runtime_error("Unknown algorithm");
```

Comment:
instead of throwing, maybe just return a default constructed KeyPairGenerationParameters() and QSKIP in that case.

Reply:
Ok!